src/Impostor.Server/Net/State/Game.Outgoing.cs

```diff
@@ -100,5 +120,27 @@ private void WriteWaitForHostMessage(IMessageWriter message, bool clear, IClient
         {
             Message12WaitForHostS2C.Serialize(message, clear, Code, player.Client.Id);
         }
+
+        private ValueTask SendObjectSpawn(InnerNetObject obj, int? targetClientId = null)
+        {
+            using var writer = StartGameData(targetClientId);
+            writer.StartMessage(GameDataTag.SpawnFlag);
+            writer.WritePacked(SpawnableObjectIds[obj.GetType()]);
+            writer.WritePacked(obj.OwnerId);
+            writer.Write((byte)obj.SpawnFlags);
+
+            var components = obj.GetComponentsInChildren<InnerNetObject>();
+            writer.WritePacked(components.Count);
+            foreach (var comp in components)
+            {
+                writer.WritePacked(obj.NetId);
+                writer.StartMessage(1);
+                comp.SerializeAsync(writer, true);
```

Comment:
```suggestion
                await comp.SerializeAsync(writer, true);
```

Reply:
... what a footgun, why don't static analyzers pick up unawaited futures?

Reply:
You would get [`CS4014`](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs4014) if `SendObjectSpawn` was async. I guess this is yet another reason to avoid tail returning tasks vs just awaiting them.